src/smt/theory_str.cpp

```diff
@@ -1880,6 +1880,16 @@ namespace smt {
             SASSERT(axiom1);
             assert_axiom(axiom1);
         }
+
+        // axiom 2: The only (str.from-int N) that starts with a "0" is "0".
+        expr_ref zero(mk_string("0"), m);
+        // let (the result starts with a "0") be p
+        expr_ref p(u.str.mk_prefix(zero, ex), m);
```

Comment:
I would suggest using more descriptive variable names, as it will greatly help the next time this code needs to be debugged or modified.